src/Analysis/GWASV2/Shared/Covariates/CustomDichotomousCovariates.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import CohortSelect from '../../CohortSelect/CohortSelect';
+import ACTIONS from '../StateManagement/Actions';
+import '../../../GWASUIApp/GWASUIApp.css';
+
+const CustomDichotomousCovariates = ({ dispatch, setMode, type }) => {
+  const [firstPopulation, setFirstPopulation] = useState(undefined);
+  const [secondPopulation, setSecondPopulation] = useState(undefined);
+  const [providedName, setProvidedName] = useState('');
+
+  const handleDichotomousSubmit = () => {
+    const dichotomous = {
+      variable_type: 'custom_dichotomous',
+      cohort_ids: [
+        firstPopulation.cohort_definition_id,
+        secondPopulation.cohort_definition_id,
+      ],
+      provided_name: providedName,
+    };
+    dispatch(
+      type === 'outcome'
```

Comment:
I'm not sure I agree with this model. Ideally `CustomDichotomousCovariates` component is a generic component that is good at creating a custom dichotomous variable and informing its caller/state about the newly created item. This part of the code tries to implement all dispatching logic inside, while I think the custom dispatch logic (i.e. what to do with the newly created item) is a concern of the parent component.

Reply:
I think we'll have to fix this in a next ticket, since this introduces a mixed model that is confusing IMO. Now we have custom dispatch logic implemented in different layers...